### PR TITLE
Fix CodeAction telemetry

### PR DESCRIPTION
**Pull request by datho7561**
Fixes the following bugs:
- Some code actions would error out instead of applying
- Some code actions would have an unknown id, despite being given an id on the language server side
- Code actions weren't properly linked to the error they fix

Signed-off-by: David Thompson <davthomp@redhat.com>

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -100,34 +104,33 @@ async function doActivate(context: ExtensionContext) {
...
               return;
             }
           }
-          if (codeAction.edit) {
-            if (codeAction.data.id) {
-              codeActionId = codeAction.data.id;
-            }
-            const edit = await languageClient.protocol2CodeConverter.asWorkspaceEdit(codeAction.edit);
-            await workspace.applyEdit(edit);
+          if (vsCodeAction.edit) {
```
Comment (rgrunber) [inline src/extension.ts:126]: I would just document for the logic that we follow https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#codeAction .

```
/**
	 * A command this code action executes. If a code action
	 * provides an edit and a command, first the edit is
	 * executed and then the command.
	 */
```

File: src/extension.ts [hunk 2]
```diff
@@ -100,34 +104,33 @@ async function doActivate(context: ExtensionContext) {
...
+            await workspace.applyEdit(vsCodeAction.edit);
           }
-          if (codeAction.command) {
+          if (vsCodeAction.command) {
             if (!codeActionId) {
```
Comment (rgrunber) [inline src/extension.ts:130]: Are there any cases where we may not have an id for the code action, or is this just a sanity check / future proofing down the road ?

Comment (datho7561) [reply src/extension.ts:130]: I double checked, and it looks like I've named all the code actions that just run a command. So it looks like this is for future proofing.

**Review by rgrunber — APPROVED** [review]
I think this should be merged. This definitely fixes the existing state of code actions (which is broken for some).

**Pull request merged**